### PR TITLE
Sketcher: continuous mode improvements

### PR DESCRIPTION
**Pull request by abdullahtahiriyo**
As per:
https://forum.freecad.org/viewtopic.php?p=716863#p716863

Problems:
* Continuous mode is not well handled. Eg. drawing a rectangle, if I press Esc/RMB after clicking the first point (and before clicking the second), the rectangle feature is just left, instead of just canceling the current rectangle and be ready for a new one.
* I got some cases where pressing Esc does nothing, but I'm not able to reproduce reliably, so will eventually come back later on this one.

Solution:
- ESC or RMB will reset the handler to the first state machine state if not in the first state. If already in the first state, it terminates the handler.

### CODE REVIEW

**Comment by abdullahtahiriyo** [comment]
@PaddleStroke 
@0penBrain 

This PR needs testing. See if this meets the requirements.

If you look at the code, there is hack in there. I do not like it, but I do not know exactly what is preventing ViewProviderSketch from receiving keyboard events after reset. I would be grateful if a real solution is found.

**Comment by PaddleStroke** [comment]
> ESC or RMB will reset the handler to the first state machine state if not in the first state. If already in the first state, it terminates the handler.

This is a good solution.

File: src/Mod/Sketcher/Gui/DrawSketchController.h [hunk 1]
```diff
@@ -564,6 +564,10 @@ class DrawSketchController
         onViewIndexWithFocus = 0;
 
         configureOnViewParameters();
+
+        if (init) {
+            handler->moveCursorToSketchPoint(prevCursorPosition);
```
Comment (PaddleStroke) [inline src/Mod/Sketcher/Gui/DrawSketchController.h:569]: The bug described in #11307 (see https://youtu.be/24UYGO5PgjU) happens here. This moveCursorToSketchPoint is buggy

Comment (PaddleStroke) [reply src/Mod/Sketcher/Gui/DrawSketchController.h:569]: Besides the bug in moveCursorToSketchPoint, is this necessary here? Why are you doing this?

Comment (abdullahtahiriyo) [reply src/Mod/Sketcher/Gui/DrawSketchController.h:569]: Replied in the other PR.

**Pull request merged**